src/basic/env-util.c

```diff
@@ -183,36 +183,34 @@ static int env_append(char **r, char ***k, char **a) {
         return 0;
 }
 
-char **strv_env_merge(size_t n_lists, ...) {
+char** _strv_env_merge(char **first, ...) {
         _cleanup_strv_free_ char **ret = NULL;
-        size_t n = 0;
         char **l, **k;
         va_list ap;
 
         /* Merges an arbitrary number of environment sets */
 
-        va_start(ap, n_lists);
-        for (size_t i = 0; i < n_lists; i++) {
-                l = va_arg(ap, char**);
+        size_t n = strv_length(first);
+
+        va_start(ap, first);
+        while ((l = va_arg(ap, char**)) != (char**)SIZE_MAX)
```

Comment:
POINTER_MAX

Reply:
i really don't like this style of cramping assignment and test into one subexpression like this